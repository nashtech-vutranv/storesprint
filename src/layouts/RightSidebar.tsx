import {useContext, useEffect, useRef, useCallback} from 'react'
import {Link} from 'react-router-dom'
import SimpleBar from 'simplebar-react'
import {ThemeCustomizer} from '../components'
import {GlobalContext} from '../store/GlobalContext'
import {hideRightSidebar} from '../store/actions'

const RightSideBar = () => {
  const rightBarNodeRef = useRef<HTMLDivElement>(null)

  const {
    state: {
      layout: {showRightSidebar},
    },
    dispatch: {layout: layoutDispatch},
  } = useContext(GlobalContext)

  /**
   * Handles the close
   */
  const handleClose = (e: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => {
    e.preventDefault()
    layoutDispatch(hideRightSidebar())
  }

  /**
   * Handle the click anywhere in doc
   */
  const handleOtherClick = useCallback(
    (e: any) => {
      if (showRightSidebar) {
        if (
          rightBarNodeRef &&
          rightBarNodeRef.current &&
          rightBarNodeRef.current.contains(e.target)
        ) {
          return
        } else {
          layoutDispatch(hideRightSidebar())
        }
      }
    },
    [rightBarNodeRef, showRightSidebar, layoutDispatch]
  )

  useEffect(() => {
    document.addEventListener('mousedown', handleOtherClick, false)
    return () => {
      document.removeEventListener('mousedown', handleOtherClick, false)
    }
  })

  const manageBodyClass = (cssClass: string, action = 'toggle') => {
    switch (action) {
      case 'add':
        if (document.body) document.body.classList.add(cssClass)
        break
      case 'remove':
        if (document.body) document.body.classList.remove(cssClass)
        break
      default:
        if (document.body) document.body.classList.toggle(cssClass)
        break
    }

    return true
  }

  useEffect(() => {
    manageBodyClass('end-bar-enabled', showRightSidebar ? 'add' : 'remove')
  }, [showRightSidebar])

  useEffect(() => {
    const endBarToggleEl = document.getElementsByClassName(
      'end-bar-toggle float-end'
    )[0] as any

    if (endBarToggleEl) {
      endBarToggleEl.setAttribute('aria-label', 'endbar-toggle')
    }
  }, [])

  return (
    <>
      <div className='end-bar' ref={rightBarNodeRef}>
        <div className='rightbar-title'>
          <Link
            to='#'
            className='end-bar-toggle float-end'
            onClick={handleClose}
          >
            <i className='dripicons-cross noti-icon'></i>
          </Link>
          <h5 className='m-0'>Settings</h5>
        </div>

        <SimpleBar
          style={{maxHeight: '100%', zIndex: 10000}}
          timeout={500}
          scrollbarMaxSize={320}
        >
          <div className='rightbar-content h-100'>
            {' '}
            <ThemeCustomizer />
          </div>
        </SimpleBar>
      </div>

      <div className='rightbar-overlay'></div>
    </>
  )
}

export default RightSideBar
