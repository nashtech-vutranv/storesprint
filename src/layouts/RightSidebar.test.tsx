import {render} from '@testing-library/react'
import '@testing-library/jest-dom'
import {MemoryRouter} from 'react-router-dom'
import RightSidebar from './RightSidebar'

test('renders RightSidebar', async () => {
  const tree = render(
    <MemoryRouter>
      <RightSidebar />
    </MemoryRouter>
  )
  expect(tree).toMatchSnapshot()
})
