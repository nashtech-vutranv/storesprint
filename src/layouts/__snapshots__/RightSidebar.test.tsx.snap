// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders RightSidebar 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="end-bar"
      >
        <div
          class="rightbar-title"
        >
          <a
            aria-label="endbar-toggle"
            class="end-bar-toggle float-end"
            href="/"
          >
            <i
              class="dripicons-cross noti-icon"
            />
          </a>
          <h5
            class="m-0"
          >
            Settings
          </h5>
        </div>
        <div
          data-simplebar="init"
          style="max-height: 100%; z-index: 10000;"
        >
          <div
            class="simplebar-wrapper"
          >
            <div
              class="simplebar-height-auto-observer-wrapper"
            >
              <div
                class="simplebar-height-auto-observer"
              />
            </div>
            <div
              class="simplebar-mask"
            >
              <div
                class="simplebar-offset"
                style="right: 0px; bottom: 0px;"
              >
                <div
                  aria-label="scrollable content"
                  class="simplebar-content-wrapper"
                  role="region"
                  style="height: auto; overflow-x: hidden; overflow-y: hidden;"
                  tabindex="0"
                >
                  <div
                    class="simplebar-content"
                  >
                    <div
                      class="rightbar-content h-100"
                    >
                       
                      <div
                        class="p-3"
                      >
                        <h5>
                          layout_color_scheme
                        </h5>
                        <hr
                          class="mt-1"
                        />
                        <div
                          class="form-check form-switch mb-1 form-check"
                        >
                          <input
                            aria-label="toggle-left-sidebar-color"
                            class="form-check-input"
                            id="light-mode"
                            name="layout-color"
                            type="radio"
                            value="light"
                          />
                          <label
                            class="form-check-label"
                            for="vertical-layout"
                          >
                            layout_color_scheme_light_mode
                          </label>
                        </div>
                        <div
                          class="form-check form-switch mb-1 form-check"
                        >
                          <input
                            aria-label="toggle-left-sidebar-color"
                            checked=""
                            class="form-check-input"
                            id="dark-mode"
                            name="layout-color"
                            type="radio"
                            value="dark"
                          />
                          <label
                            class="form-check-label"
                            for="horizontal-layout"
                          >
                            layout_color_scheme_dark_mode
                          </label>
                        </div>
                        <h5
                          class="mt-4"
                        >
                          layout_left_side_color
                        </h5>
                        <hr
                          class="mt-1"
                        />
                        <div
                          class="form-check form-switch mb-1 form-check"
                        >
                          <input
                            aria-label="toggle-left-sidebar-color"
                            class="form-check-input"
                            id="brand-check"
                            name="theme"
                            type="radio"
                            value="default"
                          />
                          <label
                            class="form-check-label"
                            for="brand-check"
                          >
                            layout_left_side_color_default
                          </label>
                        </div>
                        <div
                          class="form-check form-switch mb-1 form-check"
                        >
                          <input
                            aria-label="toggle-left-sidebar-color"
                            class="form-check-input"
                            id="light-check"
                            name="theme"
                            type="radio"
                            value="light"
                          />
                          <label
                            class="form-check-label"
                            for="light-check"
                          >
                            layout_left_side_color_light
                          </label>
                        </div>
                        <div
                          class="form-check form-switch mb-1 form-check"
                        >
                          <input
                            aria-label="toggle-left-sidebar-color"
                            checked=""
                            class="form-check-input"
                            id="dark-check"
                            name="theme"
                            type="radio"
                            value="dark"
                          />
                          <label
                            class="form-check-label"
                            for="dark-check"
                          >
                            layout_left_side_color_dark
                          </label>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="simplebar-placeholder"
              style="width: 0px; height: 0px;"
            />
          </div>
          <div
            class="simplebar-track simplebar-horizontal"
            style="visibility: hidden;"
          >
            <div
              class="simplebar-scrollbar"
              style="width: 0px; display: none;"
            />
          </div>
          <div
            class="simplebar-track simplebar-vertical"
            style="visibility: hidden;"
          >
            <div
              class="simplebar-scrollbar"
              style="height: 0px; display: none;"
            />
          </div>
        </div>
      </div>
      <div
        class="rightbar-overlay"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="end-bar"
    >
      <div
        class="rightbar-title"
      >
        <a
          aria-label="endbar-toggle"
          class="end-bar-toggle float-end"
          href="/"
        >
          <i
            class="dripicons-cross noti-icon"
          />
        </a>
        <h5
          class="m-0"
        >
          Settings
        </h5>
      </div>
      <div
        data-simplebar="init"
        style="max-height: 100%; z-index: 10000;"
      >
        <div
          class="simplebar-wrapper"
        >
          <div
            class="simplebar-height-auto-observer-wrapper"
          >
            <div
              class="simplebar-height-auto-observer"
            />
          </div>
          <div
            class="simplebar-mask"
          >
            <div
              class="simplebar-offset"
              style="right: 0px; bottom: 0px;"
            >
              <div
                aria-label="scrollable content"
                class="simplebar-content-wrapper"
                role="region"
                style="height: auto; overflow-x: hidden; overflow-y: hidden;"
                tabindex="0"
              >
                <div
                  class="simplebar-content"
                >
                  <div
                    class="rightbar-content h-100"
                  >
                     
                    <div
                      class="p-3"
                    >
                      <h5>
                        layout_color_scheme
                      </h5>
                      <hr
                        class="mt-1"
                      />
                      <div
                        class="form-check form-switch mb-1 form-check"
                      >
                        <input
                          aria-label="toggle-left-sidebar-color"
                          class="form-check-input"
                          id="light-mode"
                          name="layout-color"
                          type="radio"
                          value="light"
                        />
                        <label
                          class="form-check-label"
                          for="vertical-layout"
                        >
                          layout_color_scheme_light_mode
                        </label>
                      </div>
                      <div
                        class="form-check form-switch mb-1 form-check"
                      >
                        <input
                          aria-label="toggle-left-sidebar-color"
                          checked=""
                          class="form-check-input"
                          id="dark-mode"
                          name="layout-color"
                          type="radio"
                          value="dark"
                        />
                        <label
                          class="form-check-label"
                          for="horizontal-layout"
                        >
                          layout_color_scheme_dark_mode
                        </label>
                      </div>
                      <h5
                        class="mt-4"
                      >
                        layout_left_side_color
                      </h5>
                      <hr
                        class="mt-1"
                      />
                      <div
                        class="form-check form-switch mb-1 form-check"
                      >
                        <input
                          aria-label="toggle-left-sidebar-color"
                          class="form-check-input"
                          id="brand-check"
                          name="theme"
                          type="radio"
                          value="default"
                        />
                        <label
                          class="form-check-label"
                          for="brand-check"
                        >
                          layout_left_side_color_default
                        </label>
                      </div>
                      <div
                        class="form-check form-switch mb-1 form-check"
                      >
                        <input
                          aria-label="toggle-left-sidebar-color"
                          class="form-check-input"
                          id="light-check"
                          name="theme"
                          type="radio"
                          value="light"
                        />
                        <label
                          class="form-check-label"
                          for="light-check"
                        >
                          layout_left_side_color_light
                        </label>
                      </div>
                      <div
                        class="form-check form-switch mb-1 form-check"
                      >
                        <input
                          aria-label="toggle-left-sidebar-color"
                          checked=""
                          class="form-check-input"
                          id="dark-check"
                          name="theme"
                          type="radio"
                          value="dark"
                        />
                        <label
                          class="form-check-label"
                          for="dark-check"
                        >
                          layout_left_side_color_dark
                        </label>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="simplebar-placeholder"
            style="width: 0px; height: 0px;"
          />
        </div>
        <div
          class="simplebar-track simplebar-horizontal"
          style="visibility: hidden;"
        >
          <div
            class="simplebar-scrollbar"
            style="width: 0px; display: none;"
          />
        </div>
        <div
          class="simplebar-track simplebar-vertical"
          style="visibility: hidden;"
        >
          <div
            class="simplebar-scrollbar"
            style="height: 0px; display: none;"
          />
        </div>
      </div>
    </div>
    <div
      class="rightbar-overlay"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
